Merchant/App/Settings/SettingsViewController.swift

```diff
@@ -97,8 +97,15 @@ class SettingsViewController: BDCViewController {
         destinationAddressTextField.text = destination
     }
     
-    func onGetCurrency(_ currency: String) {
-        selectedCurrencyLabel.text = currency
+    func onGetCurrency(_ currency: StoreCurrency) {
+        var text = currency.name
+        let rate = GetRateInteractor().getRate(withCurrency: currency)
+        if let rate = rate?.rate {
+            if (rate > 0) {
```

Comment:
You can combine these 2 if statements: if let rate = rate?.rate, rate > 0 { ... }